src/core/qgsgeometry.cpp

```diff
@@ -4632,6 +4752,34 @@ void QgsGeometry::translateVertex( QgsWkbPtr &wkbPtr, double dx, double dy, bool
     wkbPtr += sizeof( double );
 }
 
+void QgsGeometry::transformVertex( QgsWkbPtr &wkbPtr, const QTransform& trans, bool hasZValue )
+{
+  double x, y, rotated_x, rotated_y;
+
+  QgsWkbPtr tmp = wkbPtr;
+
+  memcpy( &x, tmp, sizeof( double ) );
+  tmp += sizeof( double );
+  memcpy( &y, tmp, sizeof( double ) );
+  tmp += sizeof( double );
+
+  QPointF ip(x,y);
+  QPointF op = trans.map( ip );
```

Comment:
@strk might be a bit faster to use QTransform::map ( qreal x, qreal y, qreal * tx, qreal * ty ) const here rather than creating these two temporary QPointF objects.